crates/sui-core/src/authority_client.rs

```diff
@@ -90,7 +93,7 @@ pub trait AuthorityAPI {
 
 #[derive(Clone)]
 pub struct NetworkAuthorityClient {
-    client: ValidatorClient<Channel>,
+    client: SuiResult<ValidatorClient<Channel>>,
```

Comment:
Probably better to just use Option instead of SuiResult?

Reply:
The idea is that this propagates the actual error though